File: 7-ReverseInteger/7-ReverseInteger.py
# Last updated: 8/19/2026, 10:42:46 AM
class Solution:
    def reverse(self, x: int) -> int:
        if x>0:
            ans=int(str(x)[::-1])
        else:
            ans=-int(str(-x)[::-1])
        if -2**31 <= ans <= 2**31 - 1:
            return ans
        else:
            return 0
    
    
    

File: 7-ReverseInteger/7-ReverseInteger.java
// Last updated: 8/19/2026, 10:42:40 AM
class Solution {
    public int reverse(int x) {
        long rev=0;
        while(x!=0)
            {
            rev=(rev*10)+(x%10);
            x/=10;
            if((rev<- 2147483647) || (rev> 2147483647))
                return 0;

        }
        return (int)rev;
    }
}